src/check_type.cpp

```diff
@@ -370,7 +373,9 @@ gb_internal void add_polymorphic_record_entity(CheckerContext *ctx, Ast *node, T
 	rw_mutex_lock(&ctx->info->gen_types_mutex);
 	auto *found_gen_types = map_get(&ctx->info->gen_types, original_type);
 	if (found_gen_types) {
-		rw_mutex_lock(&found_gen_types->mutex);
+		if (!local_gen_types_locked) {
+			rw_mutex_lock(&found_gen_types->mutex);
+		}
 		array_add(&found_gen_types->types, e);
 		rw_mutex_unlock(&found_gen_types->mutex);
```

Comment:
Does this require `local_gen_types_locked` too?

Reply:
Sorry, I don't understand. This is the only place it is checked and it's required to avoid locking twice.